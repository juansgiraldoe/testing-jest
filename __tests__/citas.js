import Citas from "../js/classes/Citas";

describe('Probar la clase de citas', () => {
  
  const citas = new Citas();
  const id= Date.now();

  test('Agregar una nueva cita', () => {
    const citaObj = {
      mascota: 'Honey',
      propietario: 'Juan',
      telefono: '310',
      fecha: '14-05-2024',
      hora:'10:30',
      sintomas: 'Come mucho.'
    };
    citaObj.id = id;
    citas.agregarCita(citaObj);

    //Test.
    expect(citas).toMatchSnapshot();
  });

  test('Actualizar cita', () => {
    const citaActualizada = {
      id,
      mascota: 'Honey velez',
      propietario: 'Melissa',
      telefono: '31055',
      fecha: '14-05-2024',
      hora:'08:30',
      sintomas: 'Come y duerme mucho.'
    };
    citas.editarCita(citaActualizada);

    expect(citas).toMatchSnapshot();
  });

  test('Eliminar cita', () => {
    citas.eliminarCita(id);

    expect(citas).toMatchSnapshot();
  });
});
