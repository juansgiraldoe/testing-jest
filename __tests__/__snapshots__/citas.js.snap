// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probar la clase de citas Actualizar cita 1`] = `
Citas {
  "citas": [
    {
      "fecha": "14-05-2024",
      "hora": "08:30",
      "id": 1710456718744,
      "mascota": "Honey velez",
      "propietario": "Melissa",
      "sintomas": "Come y duerme mucho.",
      "telefono": "31055",
    },
  ],
}
`;

exports[`Probar la clase de citas Agregar una nueva cita 1`] = `
Citas {
  "citas": [
    {
      "fecha": "14-05-2024",
      "hora": "10:30",
      "id": 1710456718744,
      "mascota": "Honey",
      "propietario": "Juan",
      "sintomas": "Come mucho.",
      "telefono": "310",
    },
  ],
}
`;

exports[`Probar la clase de citas Eliminar cita 1`] = `
Citas {
  "citas": [],
}
`;
